src/main/java/de/tum/in/www1/artemis/service/AssessmentService.java

```diff
@@ -109,8 +109,22 @@ public Result updateAssessmentAfterComplaint(Result originalResult, Exercise exe
 
         // Update the result that was complained about with the new feedback
         originalResult.updateAllFeedbackItems(assessmentUpdate.getFeedbacks());
-        Double calculatedScore = calculateTotalScore(originalResult.getFeedbacks());
-        return submitResult(originalResult, exercise, calculatedScore);
+        if (exercise instanceof ProgrammingExercise) {
+            double points = ((ProgrammingAssessmentService) this).calculateTotalScore(originalResult);
```

Comment:
Do we need the exercise type case distinction here? If `ProgrammingAssessmentService` inherits from `AssessmentService`, you should override `updateAssessmentAfterComplaint` and call it on the child class.

Reply:
I cannot override the method as there is no `calculateTotalScore(Result result)` in `AssessmentService` only for programming exercises we have a special way on calculating the points the student has :)

Reply:
I understand your point, but it is basically then code duplication except of the last part. I am not quite sure it this is then better maintainable?